### PR TITLE
[WIP] LoopKernel

### PR DESCRIPTION
**Pull request by Krovatkin**

### CODE REVIEW

File: test/cpu_fusion.cpp [hunk 1]
```diff
@@ -1448,6 +1448,231 @@ TEST(cpu_fusion, backwards_maxpool_with_indices_n4_c1_hw4_2x2_max)
...
+    shared_ptr<runtime::Tensor> b = backend->create_tensor(element::f32, shapeA);
+    shared_ptr<runtime::Tensor> result_relu = backend->create_tensor(element::f32, shapeA);
+    shared_ptr<runtime::Tensor> result_add = backend->create_tensor(element::f32, shapeA);
+
+    vector<float> dataA{-1, 4, -1, 4};
+    vector<float> dataB{0, 4, 0, 4};
+    copy_data(a, dataA);
+    copy_data(b, dataB);
+    vector<float> expected_relu{0, 4, 0, 4};
+    vector<float> expected_add{4, 4, 4, 4};
```
[the remark this answers is not in the document]
Comment (rkimballn1) [reply test/cpu_fusion.cpp]: This is a good comment. Klocwork will complain about this too.

File: src/ngraph/runtime/cpu/builder/loop_kernel.cpp [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+                    //a subgraph
+                    if (op->get_outputs().size() > 1)
+                    {
+                        throw ngraph_error("no multi-output ops in a LoopKernel");
+                    }
+                    halide_functions[op->get_output_tensor_ptr()->get_name()] =
+                        halide::generators.at(TI(*op))(inputs);
+                }
+
+                auto& functors = external_function->get_functors();
```
[the remark this answers is not in the document]
Comment (rkimballn1) [reply src/ngraph/runtime/cpu/builder/loop_kernel.cpp]: functors is actually used. -1 for codacy

File: src/ngraph/codegen/CMakeLists.txt [hunk 3]
```diff
@@ -76,6 +76,9 @@ if (NGRAPH_GPU_ENABLE OR (NGRAPH_CPU_ENABLE AND NOT NGRAPH_DEX_ONLY))
     add_dependencies(codegen header_resource)
     if (NGRAPH_CPU_ENABLE)
         add_dependencies(codegen libmkldnn libeigen)
+        # --exclude-libs=ALL prevents symbols from statically-linked libraries (LLVM, in this case)
+        # from being automatically exported
+        set_property(TARGET codegen APPEND PROPERTY LINK_FLAGS "-Wl,--exclude-libs=ALL")
```
Comment (rkimballn1) [inline src/ngraph/codegen/CMakeLists.txt]: Is this really necessary? target_link_libraries PRIVATE on line 84 is supposed to make all symbols private.

File: test/cpu_fusion.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (Krovatkin) [reply test/cpu_fusion.cpp]: yeah, this is actually a real issue. I should be comparing *both* outputs.

**Comment by Krovatkin** [comment]
Closing in favour of https://github.com/NervanaSystems/ngraph/pull/2156

**Pull request closed without merging**